src/geometry/utility/geometry.cc

```diff
@@ -325,8 +334,46 @@ double Get2DHeadingAtP(const LineString3d& line_string, double p) {
 }
 
 Vector2 Get2DTangentAtP(const LineString3d& line_string, double p) {
-  const double heading = Get2DHeadingAtP(line_string, p);
-  return {std::cos(heading), std::sin(heading)};
+  // const double heading = Get2DHeadingAtP(line_string, p);
+  // return {std::cos(heading), std::sin(heading)};
+  const auto line_string_points_length = GetBoundPointsAtP(line_string, p);
+  const Vector2 d_xy{To2D(*line_string_points_length.second) - To2D(*line_string_points_length.first)};
+  return (d_xy / (To2D(line_string).length())).normalized();
+}
+
+Vector3 GetTangentAtP(const LineString3d& line_string, double p) {
+  const auto line_string_points_length = GetBoundPointsAtP(line_string, p);
+  const Vector3 d_xyz{*line_string_points_length.second - *line_string_points_length.first};
+  return (d_xyz / (line_string.length())).normalized();
+}
+
+ClosestPointResult GetClosestPoint(const Segment3d& segment, const maliput::math::Vector3& xyz) {
+  const maliput::math::Vector3 d_segment{segment.second - segment.first};
+  const maliput::math::Vector3 d_xyz_to_first{xyz - segment.first};
+  const maliput::math::Vector3 unit_vector{d_segment / d_segment.norm()};
```

Comment:
Consider using Vector::nomalized(). Also, you could use it in place and save you from declaring the variable for that purpose only.

Reply:
Right!
Done!